MIPs/mip-eth-sign.md

```diff
@@ -0,0 +1,47 @@
+---
+MIP: X
+Title: Deprecation of eth_sign
+Status: Last Call
+Stability: Deprecated
+discussions-to: https://github.com/MetaMask/metamask-improvement-proposals/discussions/41
+Author(s): Vandan Parikh(@vandan)
+Type: Maintainer
+Created: 2024-05-15
+---
+
+## Summary
+This proposal seeks to fully deprecate the `eth_sign` signature method in MetaMask due to its inherent security risks. The `eth_sign` method allows signing arbitrary data, posing a significant phishing risk. As of last year, `eth_sign` was disabled by default, but could be re-enabled in MetaMask's settings. This proposal aims to discontinue support entirely.
+
+## Motivation
+The primary motivation for this deprecation is to enhance user security. The `eth_sign` method's capability to sign arbitrary data makes it vulnerable to phishing attacks, where malicious actors can deceive users into signing harmful transactions or data. Even with `eth_sign` being disabled by default, there is still a risk that users are enticed to re-enable it when they should not. By removing support for this method entirely, we aim to reduce the attack surface and protect MetaMask users from potential threats.
```

Comment:
```suggestion
The primary motivation for this deprecation is to enhance user security. The `eth_sign` method's capability to sign arbitrary data makes it vulnerable to phishing attacks, where malicious actors can deceive users into blindly signing harmful transactions or data since the hex data being signed can't be decoded. Even with `eth_sign` being disabled by default, there is still a risk that users are enticed to re-enable it when they should not. By removing support for this method entirely, we aim to reduce the attack surface and protect MetaMask users from potential threats.
```